app/com/lucidchart/open/cashy/controllers/BrowseController.scala

```diff
@@ -148,6 +156,66 @@ class BrowseController extends AppController {
   }
 
   /**
+   * Marks a hidden asset or folder as not hidden
+   */
+  def show(bucket: String, key: String) = AuthAction.authenticatedUser { implicit user =>
+    Action { implicit request =>
+      try {
+        if (key.endsWith("/")) {
+          // It is a folder
+          FolderModel.findByKey(bucket, key) match {
+            case Some(folder) => FolderModel.updateHidden(folder.id, false)
+            case None => FolderModel.createFolder(bucket, key, false)
+          }
+        } else {
+          AssetModel.findByKey(bucket, key) match {
+            case Some(asset) => AssetModel.updateHidden(asset.id, false)
+            case None => throw new Exception("Asset does not exist")
+          }
+        }
+        Ok
+      }
+      catch {
+        case e: Exception => {
+          Logger.error("Exception when showing item", e)
+          val json = Json.stringify(Json.toJson(Map("error" -> e.getMessage)))
+          Ok(json).withHeaders("Content-Type" -> "application/json")
+        }
+      }
+    }
+  }
+
+  /**
+   * Marks a shown asset or folder as hidden
+   */
+  def hide(bucket: String, key: String) = AuthAction.authenticatedUser { implicit user =>
```

Comment:
This method is essentially identical to the show method. They should be abstracted into a private method where you pass in the value of `hidden`.